thriftpy/parser/lexer.py

```diff
@@ -198,7 +198,7 @@ def t_BOOLCONSTANT(t):
 
 
 def t_DUBCONSTANT(t):
-    r'-?\d+\.\d*(e-?\d+)?'
+    r'[+-]?\d+(?=\.|[Ee])(\.\d*)?([Ee][+-]?\d+)?'
```

Comment:
This is implemented using a lookahead assertion because the pattern becomes otherwise ambiguous relative to the INTCONSTANT rules.